pkg/prometheus/promcfg.go

```diff
@@ -100,11 +100,17 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 	}, nil
 }
 
-func getScrapeClassConfig(cpf monitoringv1.CommonPrometheusFields) (map[string]*monitoringv1.ScrapeClass, string, error) {
+func getScrapeClassConfig(p monitoringv1.PrometheusInterface, cpf monitoringv1.CommonPrometheusFields) (map[string]*monitoringv1.ScrapeClass, string, error) {
 	scrapeClasses := make(map[string]*monitoringv1.ScrapeClass, len(cpf.ScrapeClasses))
 	defaultScrapeClass := ""
 	for _, scrapeClass := range cpf.ScrapeClasses {
 		scrapeClasses[scrapeClass.Name] = &scrapeClass
+		// Validate all scrape class relabelings are correct.
+		if scrapeClass.Relabelings != nil {
```

Comment:
@simonpasquier This is where I thought it would be best but I'm not sure about the error message. Do you have any suggestions?

Reply:
Looks good to me
(nit) validateRelabelConfigs should work with nil slices too.
```suggestion
```